python/cudf/dataframe/multiindex.py

```diff
@@ -98,14 +111,11 @@ def _validate_levels_and_codes(self, levels, codes):
                                  'than maximum level size at this position')
 
     def copy(self, deep=True):
-        if hasattr(self, '_source_data'):
-            mi = MultiIndex(source_data=self._source_data)
-            if self._levels is not None:
-                mi._levels = self._levels.copy()
-            if self._codes is not None:
-                mi._codes = self._codes.copy(deep)
-        else:
-            mi = MultiIndex(self.levels.copy(), self.codes.copy(deep))
+        mi = MultiIndex(source_data=self._source_data)
```

Comment:
Do we not need to copy this here since the `reset_index` call returns by copy in the constructor?

Reply:
I deleted the conditional check and left the body of the `if` statement, so looks like it was an oversight with the original implementation. It also looks like `levels.copy()` will only copy the np.array instance, not the Series inside. Also is there a special reason it's using `np.array()` to wrap a list of Series, rather than a regular Python list?